test/main.mligo

```diff
@@ -0,0 +1,156 @@
+        let () = ASSERT.tx_success result_cbk in
+        let cbk_storage = Test.get_storage cbk_ctr in
+        let () = assert(cbk_storage = init_token_supply) in
+        Test.log("OK", cbk_storage)
+    in
+
+    (* Getting Balance of Alice *)
+    let test_get_balance_from_storage =
+        let () = Test.log("--> test_get_balance_from_storage : Alice should have init_token_balance = 1000n") in
+        let retrieved_balance : nat = FUNC.get_balance_from_storage(ant_addr, alice) in
+        let () = assert(retrieved_balance = init_token_balance) in
+        Test.log("OK, Alice :", retrieved_balance)
+    in
+    
+    (* Testing GetBalance Callback for Alice *)
+    let test_get_balance_callback = 
+        let () = Test.log("--> test_get_balance_callback : Testing GetBalance Callback for Alice") in
+        let balance_of_request = ({
+            owner = alice;
+            callback = cbk_addr;
+        } : ANTI.getBalance) in
+        let result_cbk = Test.transfer_to_contract ant_ctr (GetBalance balance_of_request) 0tez in
+        let () = ASSERT.tx_success result_cbk in
+        let cbk_storage = Test.get_storage cbk_ctr in
+        let () = assert(cbk_storage = init_token_balance) in
+        Test.log("OK, Alice :", cbk_storage)
+    in
+
+    (* OK : Transferring 100 from Alice to a random contract *)
+    let test_transfer_to_contract =
+        let () = Test.log("--> test_transfer_to_contract : Transferring 100n from Alice to a random contract") in
+        let tsfr_amount = 100n in
+        let result : test_exec_result = FUNC.transfer(ant_ctr, alice, random_contract_addr, tsfr_amount) in
+        let () = ASSERT.tx_success result in
+        ASSERT.assert_transfer_contract(ant_addr, alice, random_contract_addr, init_token_balance, tsfr_amount)
+    in
+    
+    let (ant_ctr, ant_addr, cbk_addr, cbk_ctr, alice, bob, james) = FUNC.bootstrap init_token_supply init_token_balance in
+
+    (* OK : Transferring 100 from Alice to Bob *)
+    let test_transfer_with_balance =
+        let () = Test.log("--> test_transfer_with_balance : Transferring 100n from Alice to Bob") in
+        let tsfr_amount = 100n in
+        let result : test_exec_result = FUNC.transfer(ant_ctr, alice, bob, tsfr_amount) in
+        let () = ASSERT.tx_success result in
+        ASSERT.assert_transfer_account(ant_addr, alice, bob, init_token_balance, tsfr_amount)
+    in
+    
+    (* KO : Transferring 300 from Bob to Alice *)    
+    let test_transfer_with_no_balance =
+        let () = Test.log("--> test_transfer_with_no_balance : Transferring 300 from Bob to Alice") in
+        let tsfr_amount = 300n in
+        let result : test_exec_result = FUNC.transfer(ant_ctr, bob, alice, tsfr_amount) in
+        ASSERT.string_failure result ERROR.err_not_enough_balance
+    in
+
+    let (ant_ctr, ant_addr, cbk_addr, cbk_ctr, alice, bob, james) = FUNC.bootstrap init_token_supply init_token_balance in
```

Comment:
I think this is already the case in the current test
The concurrent two tests are adjacent so that we can test a specific case : sending 100 to Bob, and trying a 300 transfer with insufficient balance back from Bob.
If we bootstrap the environment again; Bob won't have any tokens (which is the following test line 82 with James)